### PR TITLE
fix: proposal remove logic and UI

### LINKED ISSUE #5435: [Task]: Only remove proposal if not voting on it
### Task description

A user should only be able to remove a proposal if he is currently not voting on it

### Requirements

TBD

### Creation checklist

- [ ] I have assigned this task to the correct people
- [X] I have added the most appropriate labels
- [X] I have linked the correct milestone and/or project

### PR DESCRIPTION
**Pull request by jeeanribeiro**
## Summary
Fixes the proposal remove logic by not allowing the user to remove the proposal if is there any account voting for proposal.
Adds a tooltip to explain why the remove proposal button is disabled.

## Relevant Issues
closes #5435 

### Platforms
- __Desktop__
  - [ ] MacOS
  - [x] Linux
  - [ ] Windows
- __Mobile__
  - [ ] iOS
  - [ ] Android

## Checklist
- [x] I have followed the contribution guidelines for this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added or modified tests that prove my changes work as intended
- [x] I have verified that new and existing unit tests pass locally with my changes
- [x] I have verified that my latest changes pass CI workflows for testing and linting
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

File: packages/shared/components/modals/ProposalDetailsMenu.svelte [hunk 1]
```diff
@@ -28,11 +32,16 @@
     async function onMountHelper(): Promise<void> {
         try {
             isVotingForProposal = await isVotingForSelectedProposal()
+            isAnyAccountVotingForProposal = await isAnyAccountVotingForSelectedProposal()
```
Comment (Tuditi) [inline packages/shared/components/modals/ProposalDetailsMenu.svelte:35]: This is not an async function

File: packages/shared/components/modals/ProposalDetailsMenu.svelte [hunk 2]
```diff
@@ -1,15 +1,19 @@
...
     import { localize } from '@core/i18n'
+    import { isVotingForSelectedProposal, isAnyAccountVotingForSelectedProposal } from '@contexts/governance/utils'
+    import { Icon } from '@auxiliary/icon'
+    import { openPopup } from '@auxiliary/popup/actions'
 
     export let modal: Modal = undefined
 
     let isVotingForProposal: boolean
+    let isAnyAccountVotingForProposal: boolean
+    let deleteMenuItem
```
Comment (Tuditi) [inline packages/shared/components/modals/ProposalDetailsMenu.svelte:15]:
```suggestion
    let deleteMenuItem: HTMLDivElement
```

**Review by Tuditi — CHANGES_REQUESTED** [review]
There are some issues:
- We query `getParticipationOverview` a lot and it takes a long time for wallet.rs to respond. We should cache the response somehow and pass it around instead of sending an API request all the time (this should go into a separate PR).
- Related to above, remove proposal can be sometimes selected even if another account is voting due to the fact that the async/await function is slow.
- If I voted, I never see the option to stop voting anymore.

I added some minor comments, but maybe it's better to merge this PR first, so there are no conflicts for `isVotingForProposal`. https://github.com/iotaledger/firefly/pull/5501

**Review by Tuditi — CHANGES_REQUESTED** [review]
Let's revisit this once we have the current votes available in Firefly. I cannot get the stop voting button to show and I am able to remove the vote proposal from certain accounts while not from others.

Issue to be closed before this one: https://github.com/iotaledger/firefly/issues/5495

**Comment by jeeanribeiro** [comment]
Closed until we have current counted votes per profile/account

**Pull request closed without merging**